### ISSUE TITLE
Can´t publish .net core to azure from VS2017

### ISSUE THREAD

**Issue by sturlath** [dotnet/core#768]
We have a .net core api app (created in VS15 but ported to VS17). 

After we ported it we managed to publish the app to azure just fine.

But when we tried again last week we just **can´t publish the app** from VS17.

![image](https://user-images.githubusercontent.com/2386572/28502210-ed8e41da-6fdc-11e7-8b15-db176d455c26.png)

But there is **just one** error and it tells us nothing. We have search for solution for 3 days and are stuck.

**The VS error window**
> 4>------ Publish started: Project: MyService, Configuration: Release Any CPU ------
> All build submissions in a build **must use project instances originating from the same project collection**.
> 
> 4>Publish failed due to build errors. Check the error list for more details.
> ========== Build: 3 succeeded, 0 failed, 4 up-to-date, 0 skipped ==========
> ========== Publish: 0 succeeded, 1 failed, 0 skipped ==========

We just can´t find out what "_must use project instances originating from the same project collection_" is telling us and verbose logging returns nothing that helps.

The thing is **we can build and publish it _locally_** and move it to our production (local IIS server) using e.g this command

> dotnet publish --configuration Release --version-suffix 45 --output bin/output/

But there is no way to publish to azure (through VS2017 publish menu), we have tried everything we can think of.  Even tried to use ftp (but found no docs on  where to put the code).

The only thing we think that could be connected to the issue is that we have 2 API projects in the solution (sharing few common projects). **Could that explain anything**?

**So**
Any ideas?
Something we can try out? 

**## Here is debug information that hopefully can help**

**MyService.csproj**
```
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup Label="Globals">
    <SccProjectName>SAK</SccProjectName>
    <SccProvider>SAK</SccProvider>
    <SccAuxPath>SAK</SccAuxPath>
    <SccLocalPath>SAK</SccLocalPath>
  </PropertyGroup>

  <PropertyGroup>
    <TargetFramework>net461</TargetFramework>
    <NoWarn>$(NoWarn);1591</NoWarn>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
    <PreserveCompilationContext>true</PreserveCompilationContext>
    <AssemblyName>MyService</AssemblyName>
    <OutputType>Exe</OutputType>
    <PackageId>MyService</PackageId>
  </PropertyGroup>

  <ItemGroup Condition=" '$(TargetFramework)' == 'net461' ">
    <Reference Include="System" />
    <Reference Include="Microsoft.CSharp" />
  </ItemGroup>
</Project>
```

**dotnet --info**
.NET Command Line Tools (1.0.4)

Product Information:
 Version:            1.0.4
 Commit SHA-1 hash:  af1e6684fd

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\1.0.4

**Comment by leecow** [comment]
@morrisjoe - can you take a look a the above?

**Comment by morrisjoe** [comment]
@sturlath can you please try if this repro with .NET Core SDK 2.0 or not? You can install from the daily build location at https://github.com/dotnet/cli/tree/release/2.0.0

**Comment by MovGP0** [comment]
I have exactly the same issue. Also created a new project and get the exact same error.

**Comment by GudjonGunnlaugsson** [comment]
@morrisjoe I'm a coworker of @sturlath been trying to get the publish to work while he is on vacation. In the CLI the publish works fine and maybe my solution would be to upload the files myself to azure. But the Visual studio publish still brings the same error as before.
Because the CLI works i figured i would try to setup an auto build deploy on check in. Still working on that.

**Comment by Petermarcu** [comment]
@morrisjoe any other thoughts here? Can we get someone to help take a closer look?

**Comment by vijayrkn** [comment]
@sturlath @MovGP0 
This issue is fixed in Visual Studio 2017 version 15.3

You can find the release notes with the list of issues fixed in this release.
https://www.visualstudio.com/en-us/news/releasenotes/vs2017-relnotes#15.3

VS download link - https://www.visualstudio.com/downloads

**Comment by Petermarcu** [comment]
Thanks @vijayrkn 

I'm going to close this issue now.  Let us know if you still see issues on the newer version of VS.
